lib/cap_authors_poller.rb

```diff
@@ -126,6 +126,11 @@ def queue_author_for_harvest(author, skip_message)
   end
 
   def update_existing_contributions(author, incoming_authorships)
+    # inject author identity into the authorship data
+    incoming_authorships = incoming_authorships.dup.map do |contrib|
```

Comment:
/shrug on the `dup` because it gets assigned back to the same variable name within the method, but at least avoids changing the mutable object elsewhere, although the changes are not likely to impact anything else.